### PR TITLE
[Fix] add support for headless device

### PR DESCRIPTION
**Pull request by hanrui1sensetime**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

MMDeploy is failed on headless devices.(#238 )

## Modification

we check $DISPLAY for headless devices. If headless device, we don't show the visualization.

## BC-breaking (Optional)

None.

## Use cases (Optional)

On headless device, execute tools/deploy.py as [get_started](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/get_started.md)

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by grimoire** [comment]
A better way is to disable visualize in deploy.py. Any advice @RunningLeon ?

**Comment by RunningLeon** [comment]
> A better way is to disable visualize in deploy.py. Any advice @RunningLeon ?

Disable `visuallize` will add another argument to `deploy.py`, which makes it harder to use. Maybe it is better to handled inside mmdeploy.

**Comment by hanrui1sensetime** [comment]
> > A better way is to disable visualize in deploy.py. Any advice @RunningLeon ?
> 
> Disable `visuallize` will add another argument to `deploy.py`, which makes it harder to use. Maybe it is better to handled inside mmdeploy.

I think we need not add another argument to `deploy.py` if we disable `visualize`, because for headless we only detect the $DISPLAY variable to work, which can call os package to finish.

**Review by grimoire — APPROVED** [review]
LGTM. Could add a warning when display is none.

File: tools/deploy.py [hunk 1]
```diff
@@ -247,29 +247,36 @@ def main():
...
+            target=visualize_model,
+            args=(model_cfg_path, deploy_cfg_path, [checkpoint_path],
+                  args.test_img, args.device),
+            kwargs=dict(
+                backend=Backend.PYTORCH,
+                output_file=osp.join(args.work_dir, 'output_pytorch.jpg'),
+                show_result=args.show),
+            ret_value=ret_value)
+    else:
+        logger.warning('$DISPLAY is none.')
```
Comment (RunningLeon) [inline tools/deploy.py:279]: warning message is too short to let user know what's it wrong here.

File: tools/deploy.py [hunk 2]
```diff
@@ -247,29 +247,37 @@ def main():
...
+            target=visualize_model,
+            args=(model_cfg_path, deploy_cfg_path, [checkpoint_path],
+                  args.test_img, args.device),
+            kwargs=dict(
+                backend=Backend.PYTORCH,
+                output_file=osp.join(args.work_dir, 'output_pytorch.jpg'),
+                show_result=args.show),
+            ret_value=ret_value)
+    else:
+        logger.warning('$DISPLAY is none. \
```
Comment (RunningLeon) [inline tools/deploy.py:279]: "visualize_model" has been skipped maybe because it's running on a headless device.

**Review by RunningLeon — APPROVED** [review]
LGTM when change the logging message

**Pull request merged**